client/app/bundles/course/group/reducers/index.js

```diff
@@ -1,14 +0,0 @@
-import { combineReducers } from 'redux';
-
-import notificationPopup from 'lib/reducers/notificationPopup';
-
-import groupsDialog from './groupsDialog';
-import groupsFetch from './groupsFetch';
-import groupsManage from './groupsManage';
-
-export default combineReducers({
```

Comment:
Bivan proposed to combine these reducers. I think it's okay too, since weirdly enough, these three reducers aren't necessarily independent. There are some actions that are handled by 2 reducers, making it difficult to trace. This is an anti-pattern, too.

Reply:
I believe the reducers were split based on the functionalities (i.e. data fetching, ui changes, etc). Also I think sharing actions in different reducers is a known pattern as well (ref: https://www.pluralsight.com/guides/sharing-redux-actions-and-reducers https://medium.com/@tommarren/share-actions-across-multiple-reducers-in-redux-e359a75f2242). In my opinion, it would be easier to trace since now we know what a specific action could affect different changes. Also, I think the file is too fat as well.